### DISCUSSION TITLE
Renovate appears to be stuck (Unsupported lock file: pnpm-lock.yaml)

### DISCUSSION THREAD

**Discussion by jacksonblankenship** [renovatebot/renovate#22133] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

Expected PRs to be opened with dependency updates

### What you tried so far.

I've tried waiting, but nothing appears to be happening. My dashboard says "These updates are pending" which implies they should be opened eventually, but I don't see any blockers preventing them from opening. However, I am seeing `Unsupported lock file: pnpm-lock.yaml` in the logs, and I assume that's the root cause.

Here's my configuration

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base"],
  "labels": ["dependencies"],
  "packageRules": [
    {
      "groupName": "commitlint",
      "matchPackagePatterns": ["@commitlint/*"]
    },
    {
      "groupName": "eslint",
      "matchPackagePatterns": ["eslint", "eslint-*", "@types/eslint"]
    },
    {
      "matchPackageNames": ["node", "@types/node"],
      "matchCurrentVersion": "^18.0.0",
      "allowedVersions": "^18.0.0"
    },
    {
      "groupName": "prettier",
      "matchPackagePatterns": ["prettier", "prettier-*", "@types/prettier"]
    },
    {
      "groupName": "prisma",
      "matchPackagePatterns": ["prisma", "@prisma/*"]
    },
    {
      "groupName": "react",
      "matchPackagePatterns": [
        "react",
        "react-dom",
        "@types/react",
        "@types/react-dom"
      ]
    },
    {
      "groupName": "trpc",
      "matchPackagePatterns": ["@trpc/*"]
    },
    {
      "groupName": "typescript-eslint",
      "matchPackagePatterns": ["@typescript-eslint/*"]
    }
  ]
}
```

### Relevant debug logs

https://gist.github.com/jacksonblankenship/ae671a5100d96816f772acc0461c1a1a

**Comment by rarkins** [comment, marked as answer]
Renovate does not yet support lockfile-only updates for pnpm. You can follow this feature request: #21438

**Reply by jacksonblankenship** [reply]
Thanks! Based on your feedback, I updated my config to include `"rangeStrategy": "pin"` and now everything is working perfectly fine.

**Reply by RIP21** [reply]
Unsure how pin solves it? My CI checks fail on --frozen-lockfile.
No matter pinned versions or not, it should update pnpm-lock to work.
Any other options I have? Aside from removing --frozen-lockfile which is bad.

**Comment by eiinu** [comment]
I updated my config to include `"rangeStrategy": "bump"` and it worked fine.
There may be something wrong with the default  `"rangeStrategy": "auto"`.
